### PR TITLE
Update httpMultipartBodyParser, accepting "Content-Type" in addition to "content-type"

### LINKED ISSUE #474: httpMultipartBodyParser only accepts 'content-type' in lowercase
@middy/core version 1.0.0-beta3
@middy/middy/http-multipart-body-parser version 1.0.0-beta3

I have a problem with httpMultipartBodyParser. It seems only accept `content-type` in lowercase.

This works:
```json
{
    "content-type": "multipart/form-data"
}
```

This doesn't work:
```json
{
    "Content-Type": "multipart/form-data"
}
```
and throws Error:
```
Error: Missing Content-Type
    at new Busboy (/app/node_modules/busboy/lib/main.js:23:11)
    at Busboy (/app/node_modules/busboy/lib/main.js:9:12)
    at parseMultipartData (/app/node_modules/middy/src/middlewares/httpMultipartBodyParser.js:42:14)
    at before (/app/node_modules/middy/src/middlewares/httpMultipartBodyParser.js:27:16)
    at runNext (/app/node_modules/middy/src/middy.js:70:24)
    at before (/app/node_modules/middy/src/middlewares/jsonBodyParser.js:22:5)
    at runNext (/app/node_modules/middy/src/middy.js:70:24)
    at before (/app/node_modules/middy/src/middlewares/httpHeaderNormalizer.js:85:7)
    at runNext (/app/node_modules/middy/src/middy.js:70:24)
    at before (/app/node_modules/middy/src/middlewares/httpEventNormalizer.js:11:12)
    at runNext (/app/node_modules/middy/src/middy.js:70:24)
    at before (/app/node_modules/middy/src/middlewares/warmup.js:20:7)
    at runNext (/app/node_modules/middy/src/middy.js:70:24)
    at runMiddlewares (/app/node_modules/middy/src/middy.js:91:3)
    at instance (/app/node_modules/middy/src/middy.js:163:5)
    at /app/node_modules/serverless-offline/src/ApiGateway.js:941:17
    at new Promise (<anonymous>)
    at handler (/app/node_modules/serverless-offline/src/ApiGateway.js:540:16)
    at module.exports.internals.Manager.execute (/app/node_modules/@hapi/hapi/lib/toolkit.js:41:33)
    at Object.internals.handler (/app/node_modules/@hapi/hapi/lib/handler.js:46:48)
    at exports.execute (/app/node_modules/@hapi/hapi/lib/handler.js:31:36)
    at Request._lifecycle (/app/node_modules/@hapi/hapi/lib/request.js:312:68)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at Request._execute (/app/node_modules/@hapi/hapi/lib/request.js:221:9)
```

Please, help me to check. Thank you a lot.

### PR DESCRIPTION
**Pull request by deeancy**
closes #474

### CODE REVIEW

File: packages/http-multipart-body-parser/index.js [hunk 1]
```diff
@@ -2,6 +2,16 @@ const BusBoy = require('busboy')
 const contentTypeLib = require('content-type')
 const createError = require('http-errors')
 
+const sanitizeHeaders = (headers) => {
```
Comment (dkatavic) [inline packages/http-multipart-body-parser/index.js]: this function will mutate input, where lowercase 'content-type' key header would be propagated to remaining middlewares and handler. As a user of `http-multipart-body-parser` middleware, I wouldn't expect that from the middleware. Can we change this function to just get `/content-type/i` property?

Comment (lmammino) [reply packages/http-multipart-body-parser/index.js]: I agree with this comment.

As a reference we are doing something similar to what @dkatavic is suggesting in other middlewares:

 - https://github.com/middyjs/middy/blob/1.0.0-beta/packages/http-content-negotiation/index.js#L28
 - https://github.com/middyjs/middy/blob/1.0.0-beta/packages/http-urlencode-body-parser/index.js#L13

**Comment by lmammino** [comment]
Hello @deeancy, any update on this?

**Comment by willfarrell** [comment]
@dkatavic Any update on this? Would you be able to rebase with the `master` branch?

**Comment by dkatavic** [comment]
@willfarrell I could take care of it when I got free time

**Comment by willfarrell** [comment]
Please re-open when you have time.

**Pull request closed without merging**